AnkiDroid/src/main/java/com/ichi2/libanki/Note.java

```diff
@@ -389,4 +391,31 @@ public boolean equals(Object o) {
     public int hashCode() {
         return (int) (mId ^ (mId >>> 32));
     }
+
+
+    public static class ClozeUtils {
+        private static final Pattern mClozeRegexPattern = Pattern.compile("\\{\\{c(\\d+)::");
+
+        /*
+        Returns the next index that a cloze should be inserted at, given a list of fields.
+        Per the manual, cloze references are the name of the delimiters for cloze deletions e.g. {{c1::text}}
+        The next index is index after the highest existing cloze deletion, gaps are not considered.
+         */
```

Comment:
```suggestion
        /**
         * Calculate the next number that should be used if inserting a new cloze deletion.
         * Per the manual the next number should be greater than any existing cloze deletion
         * even if there are gaps in the sequence, and regardless of existing cloze ordering
         *
         * @param fieldValues Iterable of field values that may contain existing cloze deletions
         * @return the next index that a cloze should be inserted at
         */
```

That's javadoc - it un-muddles the param and return by pulling them out in a specific format, and starts with a double-asterisk

Reply:
Wasn't aware. Thanks! LGTM